lib/router.js

```diff
@@ -155,33 +201,156 @@ Router.prototype._loadModule = function(modDef, config) {
+        // Insert the method spec into the global merged spec
+        if (scope.specRoot.paths[scope.prefixPath] && methodName
+                && !scope.specRoot.paths[scope.prefixPath][methodName]) {
+            scope.specRoot.paths[scope.prefixPath][methodName] = method;
+        }
+
+        if (node.value.methods.hasOwnProperty(methodName)) {
+            var e = new Error('Trying to re-define existing method '
+                + node.value.path + ':' + methodName);
+            e.pathspec = pathspec;
+            throw e;
+        }
+
+        // Check and add method-level security specs
+        if (Array.isArray(method.security)) {
+            node.value.security = method.security.map(function(item) {
+                return {
+                    value: item,
+                    method: methodName
+                };
+            }).concat(node.value.security || []);
+        }
+
+        var backendSetup = method && method['x-setup-handler'];
+        if (backendSetup) {
+            Array.prototype.push.apply(node.value.resources,
+                handlerTemplate.parseSetupConfig(backendSetup));
+        }
+
+        var backendRequest = method && method['x-request-handler'];
+        if (backendRequest) {
+            node.value.methods[methodName] = handlerTemplate.createHandler(backendRequest, {
+                globals: node.value.globals,
+            });
+        } else if (method.operationId) {
+            var handler = scope.operations[method.operationId];
+            if (handler) {
+                node.value.methods[methodName] = handler;
+            } else {
+                throw new Error('No known handler associated with operationId '
+                    + method.operationId);
+            }
+        }
+
+        if (method && method.parameters && node.value.methods[methodName]) {
+            node.value.methods[methodName].validator = new Validator(method.parameters);
+        }
+    });
+};
+
 /**
  * Process a Swagger path spec object
  */
-Router.prototype._handleSwaggerPathSpec = function(node, pathspec,
-        operations, specRoot, prefixPath) {
+Router.prototype._handleSwaggerPathSpec = function(node, pathspec, scope) {
     var self = this;
     if (!pathspec) {
         return P.resolve();
     }
 
-    // Load sub-specs
-    var subSpecs = pathspec['x-subspecs'];
-    if (!subSpecs) {
-        // Check if there is a single child spec
-        var subSpec = pathspec['x-subspec'];
-        if (subSpec) {
-            var specRootBasePath = specRoot.basePath || '';
-            specRoot = Object.assign({}, subSpec);
+    if (pathspec['x-subspec'] || pathspec['x-subspecs']) {
+        throw new Error('x-subspec and x-subspecs is no longer supported! '
+                + 'Use x-modules instead.');
+    }
+
+    // Load sub-spec
+    var loaderPromise = P.resolve();
+    // XXX: find a better way to detect this?
+    if (pathspec.swagger) {
+        if (pathspec.info && pathspec.info['x-is-api-root']) {
```

Comment:
The swagger version is required in any case. The flag to indicate an api root is pathspec.info['x-is-api-root'], so that's one flag. I tried to do without a flag, but didn't come up with anything clean & reliable.